native/acter/src/api/room.rs

```diff
@@ -996,10 +1018,17 @@ impl Room {
 
     pub async fn set_notification_mode(&self, new_mode: Option<String>) -> Result<bool> {
         let room = self.room.clone();
+        let my_id = self.user_id()?;
+        let client = self.core.client().clone();
         let mode = new_mode.and_then(|s| notification_mode_from_input(&s));
+
         RUNTIME
             .spawn(async move {
-                let notification_settings = room.client().notification_settings().await;
+                let permitted = room.can_user_trigger_room_notification(&my_id).await?;
+                if !permitted {
+                    bail!("No permissions to change notification mode in this room");
+                }
```

Comment:
this is not the right check. every user can always change its notifications mode, as that is only for them.
```suggestion
```

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1671/commits/df6c69236f58d8d1058a76c8d7a814351729b036